src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.implementation.BulldozeProducer;
+import core.mate.academy.implementation.ExcavatorProducer;
+import core.mate.academy.implementation.TruckProducer;
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private final BulldozeProducer bulldozerProducer = new BulldozeProducer();
+    private final ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private final TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) bulldozerProducer.get();
+        } else if (type.equals(Excavator.class)) {
+            return (List<T>) excavatorProducer.get();
+        } else if (type.equals(Truck.class)) {
+            return (List<T>) truckProducer.get();
+        }
+        return Collections.emptyList();
+    }
```

Comment:
[link](https://mate-academy.github.io/jv-program-common-mistakes/java-core/pecs/pecs.html#parameterize-machineproducer-in-such-a-way-that-any-implementation-not-using-machine-or-its-subtype-could-not-be-created)
![image](https://github.com/mate-academy/jv-pecs/assets/113326347/dac63e87-02d8-4d20-a5ea-5efdcb04f992)